PCSX2/pcsx2
# More preparation work for Qt

### Description of Changes
Adding new abstractions, moving stuff to core, getting rid of multiple sources of truth.

 - Adds new `FileSystem` wrappers which handle unicode paths on Windows, but store everything as UTF-8.
 - GameDatabase is now loaded on demand, and cached in a binary file (much faster and doesn't leak memory like yaml).
 - FPS/CPU usage is moved to a new PerformanceMetrics class instead of having it in 5 places.
 - Shaders, fonts and other resources are moved in to a `resources` directory in the exe folder, instead of being embedded in the exe. This means we can drop glib as a dependency later on for Linux, and less rc rubbish on Windows.
 - Gets rid of WX usage in CDVD and BIOS loading.

NOTE: When you're missing a BIOS, it'll no longer open the settings panel. Instead, you'll just get an error message. On the flipside, now if there's any valid BIOS file in the `bios` directory, it'll automatically use that. More idiot-proofing this way I hope..

### Rationale behind Changes
wx bad.

### Suggested Testing Steps

- Make sure game database related stuff hasn't broken. For example, game fix patches/widescreen patches.
- Make sure BIOS loading isn't broken with your existing config files.
- Usual stress tests for unicode correctness on windows.
